lib/pure/collections/sequtils.nim

```diff
@@ -991,46 +980,34 @@ template mapIt*(s: typed, op: untyped): untyped =
       strings = nums.mapIt($(4 * it))
     assert strings == @["4", "8", "12", "16"]
 
-  type OutType = typeof((
-    block:
-      var it{.inject.}: typeof(items(s), typeOfIter);
-      op), typeOfProc)
+  type
+    InType  = typeof(items(s), typeOfIter)
+    OutType = typeof((
+      block:
+        var it{.inject.}: InType;
+        op), typeOfProc)
+  # we cannot use normal overloading instead of the 'when' here, as the that
+  # force `op` to be ``typed``, which is what we don't want it to be
```

Comment:
```suggestion
  # we use `when` instead of overload resolution which would require `op` to be
  # ``typed``, allowing optimization of some `op`s to inline expressions
  # instead of requiring lambdas.
```

Bit more verbose, but I believe it accurately captures the intention (don't assume the `when` is bad) and convey the impact of mucking about with it carelessly.

Reply:
Hm, I was trying to communicate that it'd be better for there to be two overloads:
```nim
template mapIt(s: typed, op: proc) = ...
template mapIt(s: typed, op: untyped) = ...
```
instead of implementing manual overloading via `when` (which is only a workaround).

Reply:
Good thing I wrote out my interpretation.

Could also do:
```nim
  # xxx: drop the `when` by overloading `mapIt`, one where the `op` arg is of
  #      type `proc` and the other `untyped`
```

Again, all soft suggestions, not blockers.

Reply:
Yep, that is better, thanks. I've turned the comment into an `xxx` one, with an additional note that a compiler issue needs to be resolved first.